presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -168,6 +168,41 @@ public static Slice stPoint(@SqlType(DOUBLE) double x, @SqlType(DOUBLE) double y
         return serialize(geometry);
     }
 
+    @SqlNullable
+    @Description("Returns a Geometry type MultiPoint object formed from an array of points")
+    @ScalarFunction("ST_MultiPoint")
+    @SqlType(GEOMETRY_TYPE_NAME)
+    public static Slice stMultiPoint(@SqlType("array(" + GEOMETRY_TYPE_NAME + ")") Block input)
+    {
+        MultiPoint multipoint = new MultiPoint();
+        for (int i = 0; i < input.getPositionCount(); i++) {
+            Slice slice = GEOMETRY.getSlice(input, i);
+
+            // Ignore null points
```

Comment:
Use `input.isNull(i)` before fetching a slice:

```
            if (input.isNull(i)) {
                continue;
            }
```

Reply:
According to the spec, this function should fail when the array contains nulls:

```
12) The null-call type-preserving SQL-invoked constructor method ST_MultiPoint(ST_Point ARRAY, INTEGER) returns an ST_MultiPoint value with:
    a) The spatial reference system identifier set to ansrid.
    b) Using the method ST_Geometries(ST_Geometry ARRAY):
        i) the ST_PrivateDimension attribute set to 0 (zero).
        ii) the ST_PrivateCoordinateDimension attribute set to the value expression: ST_GetCoordDim(apointarray).
        iii) the ST_PrivateIs3D attribute set to the value expression: ST_GetIs3D(apointarray).
        iv) the ST_PrivateIsMeasured attribute set to the value expression: ST_GetIsMeasured(apointarray).
        v) the ST_PrivateGeometries attribute set to apointarray.
```

The `ST_Geometries(ST_Geometry ARRAY)` method is defined as:

```
CREATE METHOD ST_Geometries(ageometryarray ST_Geometry ARRAY[ST_MaxGeometryArrayElements])
    RETURNS ST_MultiPoint
    FOR ST_MultiPoint
    BEGIN
        DECLARE apointarray ST_Point ARRAY[ST_MaxGeometryArrayElements ];
        SET apointarray = CAST(ageometryarray AS ST_Point ARRAY[ST_MaxGeometryArrayElements]);
        RETURN
            CASE WHEN SELF IS NULL THEN
                NULL
            ELSE
                (SELF AS ST_GeomCollection).ST_Geometries(apointarray)
            END;
    END
```

which relies on the cast from array(st_geometry) -> array(st_point), which in turn, is defined as:

```
CREATE CAST(ST_Geometry ARRAY[ST_MaxGeometryArrayElements] AS ST_Point ARRAY[ST_MaxGeometryArrayElements])
WITH FUNCTION ST_ToPointAry(ST_Point ARRAY[ST_MaxGeometryArrayElements])
AS ASSIGNMENT
```

The implementation of `ST_ToPointAry` calls:

```
...
-- If ageometryarray is the null value or contains null elements,
-- then raise an exception.
CALL ST_CheckNulls(ageometryarray);
...
```